jni-gen/src/utils.rs

```diff
@@ -72,11 +72,10 @@ pub fn java_str_to_string(string: &JavaStr) -> Result<String> {
 }
 
 pub fn java_str_to_valid_rust_argument_name(string: &JavaStr) -> Result<String> {
-    let mut res = "arg_".to_string();
-    res.push_str(
-        &java_str_to_string(string)?
-            .replace('_', "___")
-            .replace('$', "_d_"),
-    );
-    Ok(res)
+    Ok(format!("arg_{}", java_identifier_to_rust(&java_str_to_string(string)?)))
```

Comment:
Here is a change - I made it all rely on `java_identifier_to_rust`. However, before there was a change that argument name would have 3 underscores instead of 2 and a dollar would be represented as `_d_` instead of `_dollar_`. I unified the format to double underscores `__` for an underscore and `_dollar_` for a dollar sign.